src/app/hooks/useBtcClient.ts

```diff
@@ -3,17 +3,18 @@ import { useMemo } from 'react';
 import useWalletSelector from './useWalletSelector';
 
 const useBtcClient = () => {
-  const { network, btcApiUrl } = useWalletSelector();
-  const { type, btcApiUrl: remoteBtcApiURL } = network;
-  const url = btcApiUrl || remoteBtcApiURL;
+  const { network } = useWalletSelector();
+  const { type, btcApiUrl, fallbackBtcApiUrl } = network;
+  const url = btcApiUrl;
```

Comment:
Isn't this changing the behavior a bit?

My understanding is:
- The previous implementation had btcApiUrl and network.btcApiUrl in the store
- btcApiUrl was either the custom URL that the user defined in settings, or it was undefined/null if they hadn't overridden it.
- network.btcApiUrl was the default value to use if the btcApiUrl was null (not set by user)
- If a custom URL was set by the config API call, we would override that default
- If a user had set a custom URL and then reset it, we would just make btcApiUrl null and that would result in the default being used form the network.

Theoretically at least, whether it was working, I'm not sure 😅 

The problem now is that we have no distinction between a custom URL being set and not.

Reply:
Yeah, looking at the code, I don't think the above was working 100%, but it seemed to be the intention 😅

Reply:
To clarify the above because it was a bit of a dump of stuff, I think we need to satisfy the following test criteria:

scenario 1 - I haven't set up a custom URL:
- the default URL should be used (I think this is satisfied)
- If a new default URL comes form the config endpoint, my default URL should change (I _don't_ think this is satisfied)
- if I set a custom URL and then go back to settings and reset the URL, I should be reset to the URL from the config endpoint and further updates to the config endpoint should affect my URL (I _don't_ think this is satisfied)

scenario 2 - I have set up a custom URL
- The custom URL should be used (I think this is satisfied)
- If I go to settings and reset my URL, it should take the one from the config API endpoint and not necessarily the one from Core (I _don't_ think this is satisfied)

scenario 3 - regardless of settings
- If a new default is set from the API, that should be the default used on a URL reset. If the API config URL is removed, i.e. the endpoint returns null/undefined for the custom URL, we should go back to using the Core URL as the default

Reply:
Yeah that makes sense. I was wondering about what the intention was for the btcapiurl coming from app config endpoint... as it didn't seem like it was applied from the store side

Reply:
Let me revisit it tomorrow with the scenarios including updating the api url via app config endpoint

Reply:
@victorkirov see comment below regarding the remote btcApiUrl config 🙂

Reply:
Awesome!! 🎉